staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/one_type_match/doc.go

```diff
@@ -24,13 +24,19 @@ type T1 struct {
 	// +validateTrue="field T1.S"
 	S  string `json:"s"`
 	T2 T2     `json:"t2"`
+	T3 T3     `json:"t3"`
 }
 
 type T2 struct {
 	// +validateTrue="field T2.S"
 	S string `json:"s"`
 }
 
+// This type has not validatio nand should not be generated
```

Comment:
"has no validations"

This is a good addition to the test, but I want to clarify:   Is the goal to elide validation functions for any type that has no transitive validations (which I think is what we do for defaults) or only to elide validations for non-matching types (no TypeMeta), but keep functions for all "rootTypes"?

Reply:
Good catch. I think we should to the latter. I'm split this into a new PR with full test coverage.